nowcasting_dataset/data_sources/satellite/satellite_data_source.py

```diff
@@ -56,11 +59,87 @@ def _get_time_slice(self, t0_dt: pd.Timestamp) -> xr.DataArray:
         start_dt = self._get_start_dt(t0_dt)
         end_dt = self._get_end_dt(t0_dt)
         data = self.data.sel(time=slice(start_dt, end_dt))
-
         assert type(data) == xr.DataArray
 
         return data
 
+    def get_image_pixel(
+        self, data_array: xr.DataArray, x_meters_center: Number, y_meters_center: Number
+    ) -> xr.DataArray:
+        """
+        Gets the satellite image as a square around the center
+
+        Ignores x and y coordinates as for the original satellite projection each pixel varies in
+        both its x and y distance from other pixels. See Issue 401 for more details.
+
+        This results, in 'real' spatial terms, each image covering about 2x as much distance in the
+        x direction as in the y direction.
+
+        Args:
+            data_array: DataArray to subselect from
+            x_meters_center: Center of the image x coordinate in OSGB coordinates
```

Comment:
shall we trying to do https://github.com/openclimatefix/nowcasting_dataset/issues/319 when possible
i.e x_center_osgb

Reply:
Yeah, I can change it to that